### PR TITLE
Add support for k8s 1.23

### PR DESCRIPTION
**Pull request by psy-q**
## Checklist

- [x] The PR has a meaningful title. It will be used to auto generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [ ] Update the documentation.
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [ ] Link this PR to related issues or PRs.

### CODE REVIEW

**Review by simu — COMMENTED** [review]
You'll probably at least also want to update line 38 to default to K8s 1.23 now. Also, depending on what you need, you probably want to update line 50 to a more recent kube-thanos version.

**Comment by psy-q** [comment]
Thanks, I've incorporated that and also added an OCP version that should work. The version naming scheme seems to have changed from 4.9 to 4.10, the current one is for 4.11.

File: jsonnetfile.jsonnet [hunk 1]
```diff
@@ -21,6 +22,7 @@ local kube_thanos_version_map = {
   '1.20': 'release-0.17',  // corresponds to v0.17.0
   '1.21': 'release-0.19',  // corresponds to v0.19.0
   '1.22': 'release-0.22',  // corresponds to v0.22.0
+  '1.23': 'v0.26.0',
```
Comment (simu) [inline jsonnetfile.jsonnet:25]: K8s 1.23 is OpenShift 4.10, so this should be v0.23.0 (cf. https://github.com/openshift/cluster-monitoring-operator/blob/release-4.10/jsonnet/jsonnetfile.json#L58), and OpenShift 4.11 (1.24) should be v0.26.0:

```suggestion
  '1.23': 'v0.23.0',
  '1.24': 'v0.26.0',
```

**Comment by psy-q** [comment]
It didn't seem to compile with v0.29.0 so now I'm trying to default to 0.26.0.

File: jsonnetfile.jsonnet [hunk 2]
```diff
@@ -34,7 +37,7 @@ local thanos_mixin_version =
     thanos_mixin_version_map[k8s_version]
   else
     // Use most recent version if we didn't find an entry in the map
-    thanos_mixin_version_map['1.22'];
+    thanos_mixin_version_map['1.23'];
```
Comment (simu) [inline jsonnetfile.jsonnet:40]: If you add 1.24 for the mixin, you probably want to default to 1.24 here

**Review by simu — APPROVED** [review]
LGTM overall, might make sense to add 1.24 to the thanos-mixin version map as well

**Pull request merged**